### ISSUE TITLE
Consider deprecating assert_select with a numeric arg

### ISSUE THREAD

**Issue by chancancode** [rails/rails-dom-testing#58]
Given a page like so:

```html
<h1>Users</h1>

<table>
  ...
</table>

Total: <span class="total">4781</span>
```

It's pretty reasonable to write something like this:

```ruby
get "/users"
assert_select "h1", "Users"
assert_select ".total", User.count
```

However, that would give you a very confusing error:

```
Failure:
MyApp::UsersTest#test_index:
Expected exactly 4781 elements matching ".total", found 1..
Expected: 4781
  Actual: 1
```

Given that we already support `assert_select ".total", count: User.count`, I think we should consider deprecating the numeric positional argument overload (as shown in the example) in favor of the explicit keyword argument, and potentially reclaiming the space for automatic `to_s`-ing the positional argument in the future (to make the example "just work").

**Comment by rafaelfranca** [comment]
👍

**Comment by chancancode** [comment]
@rafaelfranca can we still add the deprecation to 5.1?

**Comment by rafaelfranca** [comment]
Yes, we can. Even more that this would be a deprecation in a gem, so the deprecation would also happen in 5.0. I'm fine with it.

**Comment by chancancode** [comment]
@rafaelfranca we can (probably should) bump the gem version, then we can decide whether we need to lock it in Rails 5.0

**Comment by rafaelfranca** [comment]
yes, a minor version should be sufficient

**Comment by kaspth** [comment]
Yes please!
